plugins/osdn/subnets.go

```diff
@@ -179,11 +185,15 @@ func watchNodes(oc *OvsController, ready chan<- bool, start <-chan string) {
 							log.Errorf("Error deleting subnet for node %s, old ip %s", ev.Node.Name, sub.NodeIP)
 							continue
 						}
-						sub.NodeIP = ev.Node.IP
-						err = oc.Registry.CreateSubnet(ev.Node.Name, sub)
-						if err != nil {
-							log.Errorf("Error creating subnet for node %s, ip %s", ev.Node.Name, sub.NodeIP)
-							continue
+						if nodeErr == nil {
+							sub.NodeIP = ev.Node.IP
+							err = oc.Registry.CreateSubnet(ev.Node.Name, sub)
+							if err != nil {
+								log.Errorf("Error creating subnet for node %s, ip %s", ev.Node.Name, sub.NodeIP)
+								continue
+							}
+						} else {
+							log.Errorf("Deleting invalid node %s/%s subnet: %v", ev.Node.Name, ev.Node.IP, nodeErr)
```

Comment:
Is there a subsequent action here that I'm missing to differentiate line 177's stated intention to ignore the node vs this line's state intention to delete it?